src/BenchmarkDotNet/Toolchains/MonoAotLLVM/MonoAotLLVMBuilder.cs

```diff
@@ -21,48 +21,8 @@ public BuildResult Build(GenerateResult generateResult, BuildPartition buildPart
         {
             BuildResult buildResult = dotnetCliBuilder.Build(generateResult, buildPartition, logger);
 
-            if (buildResult.IsBuildSuccess
-                || File.Exists(generateResult.ArtifactsPaths.ExecutablePath)) // the build has failed, but produced an exe
-            {
-                RenameSharedLibaries(generateResult, logger);
-            }
-
-            if (!buildResult.IsBuildSuccess && File.Exists(generateResult.ArtifactsPaths.ExecutablePath))
-            {
-                logger.WriteLineError($"The buid has failed with following error: {buildResult.ErrorMessage}");
-                logger.WriteLine("But it has produced the executable file, so we are going to try to use it.");
-
-                return BuildResult.Success(generateResult); // we lie on purpose
-            }
-
             return buildResult;
         }
 
-        // This method is as work around because the MonoAOTCompilerTask
-        // does not generate files with the right names.
-        // In the future it may have an option to write .so/.dylib files,
-        // and we can get rid of this method.
-        public void RenameSharedLibaries(GenerateResult generateResult, ILogger logger)
-        {
-            string[] publishDirFiles = Directory.GetFiles(generateResult.ArtifactsPaths.BuildArtifactsDirectoryPath, ".dll.o");
-            string sharedObjectExtension = "";
-
-            if (RuntimeInformation.IsMacOSX())
-            {
-                sharedObjectExtension = "dylib";
-            }
-            else if (RuntimeInformation.IsLinux())
-            {
-                sharedObjectExtension = "so";
-            }
-
-            foreach (string fileName in publishDirFiles)
-            {
-                string newFileName = Path.ChangeExtension(fileName, sharedObjectExtension);
-
-                logger.WriteLine($"RenameSharedLibraries: Moving ${fileName} to {newFileName}");
-                File.Move(fileName, newFileName);
-            }
-        }
```

Comment:
after the removal of all the custom logic, this type has became just a dummy wrapper for `DotNetCliBuilder`. Could you please remove it and use `DotNetCliBuilder` directly for the AOT toolchain?

https://github.com/dotnet/BenchmarkDotNet/blob/63e28c100a42a6492d09a0b93a8a4c141061bb0d/src/BenchmarkDotNet/Toolchains/MonoAotLLVM/MonoAotLLVMToolChain.cs#L20-L22